pkg/controller/controller.go

```diff
@@ -0,0 +1,175 @@
+package controller
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	certv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	configNamespace    = "openshift-config-managed"
+	kubeletCAConfigMap = "csr-controller-ca"
+)
+
+// MachineApproverReconciler reconciles a machine-approver  object
+type MachineApproverReconciler struct {
+	client.Client
+	Mgr    ctrl.Manager
+	Config ClusterMachineApproverConfig
+}
+
+func (m *MachineApproverReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		WithOptions(options).
+		For(&certificatesv1beta1.CertificateSigningRequest{}).
+		WithEventFilter(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				cert, ok := e.Object.(*certificatesv1beta1.CertificateSigningRequest)
+				if !ok {
+					return false
+				}
+				if isApproved(cert) {
+					return false
+				}
+				return true
+			},
+		}).
+		Complete(m)
+}
+
+func (m *MachineApproverReconciler) Reconcile(req ctrl.Request) (reconcile.Result, error) {
+	csr := &certificatesv1beta1.CertificateSigningRequest{}
+	err := m.Get(context.Background(), req.NamespacedName, csr)
+	if err != nil {
+		klog.Errorf("Failed to get CSR: %w", err)
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
If you return the error, it will be logged and force a requeue, so we don't need to log here either
```suggestion
		return reconcile.Result{}, err
```